src/main/java/graphql/annotations/processor/retrievers/GraphQLAdditionalTypesHandler.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2016 Yurii Rashkovskii
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+package graphql.annotations.processor.retrievers;
+
+import graphql.GraphQLException;
+import graphql.annotations.annotationTypes.GraphQLTypeResolver;
+import graphql.annotations.processor.ProcessingElementsContainer;
+import graphql.annotations.processor.searchAlgorithms.SearchAlgorithm;
+import graphql.schema.GraphQLOutputType;
+import graphql.schema.GraphQLType;
+import org.clapper.util.classutil.ClassFilter;
+import org.clapper.util.classutil.ClassFinder;
+import org.clapper.util.classutil.ClassInfo;
+import org.clapper.util.classutil.SubclassClassFilter;
+import org.osgi.service.component.annotations.Component;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static graphql.annotations.processor.util.ObjectUtil.getAllFields;
+
+@SuppressWarnings("ConstantConditions")
+@Component(service = GraphQLAdditionalTypesHandler.class, immediate = true)
+public class GraphQLAdditionalTypesHandler {
+
+    private GraphQLInterfaceRetriever graphQLInterfaceRetriever;
+    private GraphQLObjectInfoRetriever graphQLObjectInfoRetriever;
+    private SearchAlgorithm methodSearchAlgorithm;
+    private SearchAlgorithm fieldSearchAlgorithm;
+    private GraphQLObjectInfoRetriever infoRetriever;
```

Comment:
Why do we have 2 instances of GraphQLObjectInfoRetriever here ? graphQLObjectInfoRetriever should be enough

Reply:
Will fix it

Reply:
Done